### ISSUE TITLE
[FRU] - SPIKE - Investigate maintaining existing recurring donations

### ISSUE THREAD

**Issue by danielfmiranda** [MozillaFoundation/foundation.mozilla.org#10264]
When we make the migration from the old donate site to this repo, we would like to make sure that any recurring donation subscriptions remain intact. There are subscriptions set up through both Stripe and also Braintree.

However, there is a conversation in the FRU slack channel discussing a migration from Stripe. https://mozilla.slack.com/archives/C04N6UECSTX/p1677509093123009

There is also mention of a "Migration service" in the [FundraiseUp docs](https://fundraiseup.com/docs/migration-service/). It might be worth exploring this and seeing if it would work for our Stripe/Braintree users. 

If so, we can track the status of that effort through this ticket.

If not, we are going to have to copy over the Stripe/Braintree webhook code that the old donate site used to send emails after a recurring donation has been successfully charged.

**Comment by cdanfon** [comment]
[Timeboxed to 1 hour]

**Comment by danielfmiranda** [comment]
After investigating this some more, it is confirmed that the FRU migration process is going to include Stripe **and** Braintree users, with an estimated timline/completion date coming next week.

Since these recurring donations are going to be carried out by FRU after the migration, I do not believe that it is necessary to carry over the braintree endpoint code from the old repo anymore. This is beneficial as it reduces our turnaround time for the project, and also reduces the chance of any endpoint errors while carrying over the code.

That being said, since there is still no completion date for the subscription migration, we are going to need to have a plan to keep them running in the meantime.


**Here is my proposal:**

1. We continue forward with our project of copying over the donate stack pages to this repo.
2. Once that is complete and we switch over the URL `donate.mozilla.org` to point to the foundation site, we keep the original donation site up and running on a heroku URL. We then update the webhook section in braintree to point to the this URL
![Screenshot 2023-03-08 at 20-39-41 Webhooks - Braintree Gateway](https://user-images.githubusercontent.com/18314510/223918937-baa47ee6-bb13-4f29-9791-f4a26840f363.png)
3. Once the subcription migration to FRU is complete, we can take down the old donation site.
4. We keep the old donate site repo up on github, in the event that we need it up and running again.
5. At a date TBD, we take down the no longer needed donate-site repo from github.

**Comment by danielfmiranda** [comment]
Hi @mtdenton! I have updated this ticket with my plan to move forward, I was wondering if you can please take a look and let me know if I am missing anything or if you have any questions?

Thanks!
